client/modules/ftml-wasm/tests/ftml.ts

```diff
@@ -1,191 +1,184 @@
+    )
+  })
 
-FTML("version", async () => {
-  await lib.loading
-  assert.type(lib.version(), "string")
-})
+  it("version", async () => {
+    await lib.loading
+    assert.isString(lib.version())
+  })
 
-FTML("preprocess", async () => {
-  await lib.loading
-  const str = lib.preprocess(
-    "\nApple Banana \\\nCherry\\\nPineapple \\ Grape\nBlueberry\n"
-  )
-  assert.is(str, "Apple Banana CherryPineapple \\ Grape\nBlueberry")
-})
+  it("preprocess", async () => {
+    await lib.loading
+    const str = lib.preprocess(
+      "\nApple Banana \\\nCherry\\\nPineapple \\ Grape\nBlueberry\n"
+    )
+    assert.equal(str, "Apple Banana CherryPineapple \\ Grape\nBlueberry")
+  })
 
-FTML("tokenize", async () => {
-  await lib.loading
-  const str = "//1//"
-  assert.equal(lib.tokenize(str), [
-    { token: "input-start", slice: "", span: { start: 0, end: 0 } },
-    { token: "italics", slice: "//", span: { start: 0, end: 2 } },
-    { token: "identifier", slice: "1", span: { start: 2, end: 3 } },
-    { token: "italics", slice: "//", span: { start: 3, end: 5 } },
-    { token: "input-end", slice: "", span: { start: 5, end: 5 } }
-  ])
-})
+  it("tokenize", async () => {
+    await lib.loading
+    const str = "//1//"
+    assert.deepEqual(lib.tokenize(str), [
+      { token: "input-start", slice: "", span: { start: 0, end: 0 } },
+      { token: "italics", slice: "//", span: { start: 0, end: 2 } },
+      { token: "identifier", slice: "1", span: { start: 2, end: 3 } },
+      { token: "italics", slice: "//", span: { start: 3, end: 5 } },
+      { token: "input-end", slice: "", span: { start: 5, end: 5 } }
+    ])
+  })
 
-FTML("parse", async () => {
-  await lib.loading
-  const str = "//1//"
-  assert.equal(lib.parse(str), {
-    "ast": {
-      "elements": [
-        {
-          "element": "container",
-          "data": {
-            "type": "paragraph",
-            "elements": [
-              {
-                "element": "container",
-                "data": {
-                  "type": "italics",
-                  "elements": [
-                    {
-                      "element": "text",
-                      "data": "1"
-                    }
-                  ],
-                  "attributes": {}
+  it("parse", async () => {
+    await lib.loading
+    const str = "//1//"
+    assert.deepEqual(lib.parse(str) as any, {
+      "ast": {
+        "elements": [
+          {
+            "element": "container",
+            "data": {
+              "type": "paragraph",
+              "elements": [
+                {
+                  "element": "container",
+                  "data": {
+                    "type": "italics",
+                    "elements": [
+                      {
+                        "element": "text",
+                        "data": "1"
+                      }
+                    ],
+                    "attributes": {}
```

Comment:
nitpick: attributes should be before elements

Reply:
fool. it passes tests. (I'll fix it)

although I'm curious if property order actually survives through the WASM boundary, or if property order is even a thing in Rust/WASM

Reply:
Since `HashMap` doesn't have a notion of key ordering, I highly doubt serde is able to preserve that information.